server/routes/taskList.js

```diff
@@ -111,9 +111,20 @@ module.exports = ({ prisonerService, licenceService, audit, caService }) => (rou
         return res.redirect(`/hdc/review/licence/${bookingId}`)
       }
 
-      const licenceStatus = getLicenceStatus(licence)
+      let licenceStatus = getLicenceStatus(licence)
       const { statusLabel } = getStatusLabel(licenceStatus, req.user.role)
 
+      if (req.user.role === 'CA' && licence && licenceStatus.stage !== 'ELIGIBILITY') {
```

Comment:
Does this code not seem a bit out of place? 
it seems very specific for such a generic function - handling edge cases like this in generic code is a code smell.
Lots of these kinds of changes mount up and make understanding what is going on much more difficult.

Reply:
I think there might be a few possible alternatives to handling this a bit better, let's talk about it after lunch